Bugsnag/Payload/BugsnagError.m

```diff
@@ -132,6 +128,33 @@ + (BugsnagError *)errorFromJson:(NSDictionary *)json {
     return error;
 }
 
+- (void)updateWithCrashInfoMessage:(NSString *)crashInfoMessage {
+    @try {
+        // Messages that match this pattern should override the errorClass (and errorMessage if there is enough information.)
+        NSString *pattern = @"^(Assertion failed|Fatal error|Precondition failed): ((.+): )?file .+, line \\d+\n$";
+        NSRegularExpression *regex = [[NSRegularExpression alloc] initWithPattern:pattern options:NSRegularExpressionCaseInsensitive error:nil];
+        NSArray<NSTextCheckingResult *> *matches = [regex matchesInString:crashInfoMessage options:0 range:NSMakeRange(0, crashInfoMessage.length)];
+        if (matches.count != 1 || matches[0].numberOfRanges != 4) {
+            if (!self.errorMessage.length) {
```

Comment:
Maybe put this defensive self.errorMessage setter code at the very top so that an unexpected exception in the regex part above can't leave it potentially empty?

Reply:
Addressed in https://github.com/bugsnag/bugsnag-cocoa/commit/8e832e8fb72006ae52213a7f1eb67c1b71c214a5

I could not move this code to before the regex part at it would have led to setting the `errorMessage` in cases where it should not be, so there's now additional code in the `@catch` block.